File: ShoppingM8-be/src/main/java/edu/pw/shoppingm8/product/api/dto/ProductResponseDto.java
package edu.pw.shoppingm8.product.api.dto;

import javax.persistence.EnumType;
import javax.persistence.Enumerated;
import javax.validation.constraints.NotNull;

import edu.pw.shoppingm8.product.Product;
import edu.pw.shoppingm8.product.ProductCategory;
import lombok.Builder;
import lombok.Value;

@Value
@Builder
public class ProductResponseDto {
    @NotNull
    Long id;
    @NotNull
    String name;
    @NotNull
    String unit;
    @NotNull
    @Enumerated(EnumType.STRING)
    ProductCategory category;
    @NotNull
    Double requiredAmount;
    @NotNull
    Double purchasedAmount;

    public static ProductResponseDto of(Product product) {
        return builder()
            .id(product.getId())
            .name(product.getName())
            .unit(product.getUnit())
            .category(product.getCategory())
            .requiredAmount(product.getRequiredAmount())
            .purchasedAmount(product.getPurchasedAmount())
            .build();
    }
}

File: shoppingm8_fe/lib/list/product/dto/productResponseDto.dart
import 'package:shoppingm8_fe/list/product/productCategory.dart';

class ProductResponseDto {
  final num id;
  final String name;
  final num requiredAmount;
  final num purchasedAmount;
  final String unit;
  final ProductCategory category;

  ProductResponseDto({this.id, this.name, this.requiredAmount, this.purchasedAmount, this.unit, this.category});

  factory ProductResponseDto.fromJson(Map<String, dynamic> json) {
    return ProductResponseDto(
        id: json['id'],
        name: json['name'],
        requiredAmount: json['requiredAmount'],
        purchasedAmount: json['purchasedAmount'],
        unit: json['unit'],
        category: ProductCategoryHepler.getFromString(json['category'])
    );
  }
}